src/runtime/CollectionWrappers/IterableWrapper.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class IterableWrapper<T> : IEnumerable<T>
+    {
+        protected PyObject iterObject;
+        protected PyObject pyObject;
+
+        public IterableWrapper(PyObject pyObj)
+        {
+            pyObject = pyObj;
+            iterObject = new PyObject(Runtime.PyObject_GetIter(pyObj.Handle));
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            IntPtr item;
+            while ((item = Runtime.PyIter_Next(iterObject.Handle)) != IntPtr.Zero)
```

Comment:
If `PyIter_Next` fails, this should check the reason. If it is not `StopIteration`, an exception should be passed to .NET